ha/util/conf_store.py

```diff
@@ -0,0 +1,87 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+
+"""
+Wrappeprs for using the search APIs from conf store
+"""
+
+import json
+
+from ha.core.config.config_manager import ConfigManager
+from ha.k8s_setup.const import CLUSTER_CARDINALITY_KEY, CLUSTER_CARDINALITY_NUM_NODES, CLUSTER_CARDINALITY_LIST_NODES
+from cortx.utils.log import Log
+
+class ConftStoreSearch:
+
+    def __init__(self):
+        """Init method"""
+        self._confstore = ConfigManager.get_confstore()
+
+
+    def _get_data_pods(self, index):
+        """
+        Get machine ids for data pods: returns list of machine ids.
+        """
+        machine_ids = []
+        return machine_ids
+
+    def _get_server_pods(self, index):
+        """
+        Get machine ids for server pods: returns list of machine ids.
+        """
+        machine_ids = []
+        return machine_ids
+
+    def get_cluster_cardinality(self):
+        """
+        Get number of nodes(pods) and their machine ids from confstore
+        """
+
+        cluster_cardinality_key = CLUSTER_CARDINALITY_KEY
+        if self._confstore.key_exists(cluster_cardinality_key):
+            cluster_cardinality = self._confstore.get(cluster_cardinality_key)
+            _, nodes_list = cluster_cardinality.popitem()
+            nodes_dict = json.loads(nodes_list)
+
+            #Get  number of nodes and the node list
+            num_nodes = nodes_dict[CLUSTER_CARDINALITY_NUM_NODES]
+            nodes_list = nodes_dict[CLUSTER_CARDINALITY_LIST_NODES]
+            return num_nodes, nodes_list
+
+        else:
+            raise Exception
+
+    def set_cluster_cardinality(self, index):
+        """
+        Set number of nodes(pods) and their machine ids in confstore used by HA
+        """
+
+        data_pods = self._get_data_pods(index)
+        server_pods = self._get_server_pods(index)
+
+        # Combine the lists data_pods, server_pod and find unique machine ids
+        watch_pods = data_pods + server_pods
+        watch_pods = list(set(watch_pods))
+        num_pods = len(watch_pods)
+
+        Log.info(f"Cluster cardinality: number of nodes {num_pods}, machine ids for nodes {watch_pods} ")
+
+        # Update the same to consul
+        # if KV already present, it will be modified.
+        cluster_cardinality_key = CLUSTER_CARDINALITY_KEY
+        cluster_cardinality_value = {CLUSTER_CARDINALITY_NUM_NODES: num_pods, CLUSTER_CARDINALITY_LIST_NODES : watch_pods }
+        self._confstore.update(cluster_cardinality_key, json.dumps(cluster_cardinality_value))
```

Comment:
Why we are not using the direct constant here. Any reason to create a new variable for this?

Reply:
This same constant is used in get and set functions 
So in case any change is needed in the constants used (based on the pending queries ) or in future also or for additional such queries , we only need to change in const file

Reply:
Ok.